app/src/main/java/com/example/filemanager/ListAdapter.kt

```diff
@@ -1,16 +1,25 @@
 package com.example.filemanager
 
+import android.content.Context
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
 import android.widget.TextView
+import androidx.core.content.ContextCompat
 import androidx.recyclerview.widget.RecyclerView
 
-internal class ListAdapter : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
-    private var items = listOf<ListModel>()
+internal class ListAdapter(
+    private val context: Context,
+    private val list: ArrayList<ListModel>,
+    private val itemClickListener: ItemClickListener
+) :
+    RecyclerView.Adapter<ListAdapter.ViewHolder>() {
 
     internal inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
-        var name: TextView = view.findViewById(R.id.fileName)
+        var fileName: TextView = view.findViewById(R.id.fileName)
```

Comment:
is it necessary to specify the variable type what if you just ommit it. Also please allways use `val` instead of `var` if possible. `var` should be used only when the value of the variable can be changed.

`val fileName = view.findViewById(R.id.fileName)`

Reply:
If I ommit the variable type, I get the error

Reply:
what kind of error?